src/rule-unused-fragments.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const utils = require('./utils');
+
+const getGraphQLAST = utils.getGraphQLAST;
+
+function getGraphQLFragmentNames(graphQLAst) {
+  const fragmentNames = {};
+
+  function walkAST(node) {
+    if (node.kind === 'FragmentSpread') {
+      const nameNode = node.name;
+      fragmentNames[nameNode.value] = nameNode;
+    }
+    if (node.kind === 'OperationDefinition') {
+      if (node.operation === 'mutation' || node.operation === 'subscription') {
+        return;
+      }
+      node.selectionSet.selections.forEach(selection => {
+        walkAST(selection);
+      });
+      return;
+    }
+    for (const prop in node) {
+      const value = node[prop];
+      if (prop === 'loc') {
+        continue;
+      }
+      if (value && typeof value === 'object') {
+        walkAST(value);
+      } else if (Array.isArray(value)) {
+        value.forEach(child => {
+          walkAST(child);
+        });
+      }
+    }
+  }
+
+  walkAST(graphQLAst);
+  return fragmentNames;
+}
+
+function getGraphQLFragmentDefinitionName(graphQLAst) {
+  const fragmentDefinition = graphQLAst.definitions.find(
+    definition => definition.kind === 'FragmentDefinition'
+  );
+  if (fragmentDefinition) {
+    return fragmentDefinition.name.value;
+  } else {
+    return null;
+  }
+}
+
+function isGraphQLTemplate(node) {
+  return (
+    node.tag.type === 'Identifier' &&
+    node.tag.name === 'graphql' &&
+    node.quasi.quasis.length === 1
+  );
+}
+
+function rule(context) {
+  let foundImportedModules = [];
+  let templateLiterals = [];
+
+  return {
+    Program(_node) {
+      foundImportedModules = [];
+      templateLiterals = [];
+    },
+    'Program:exit'(_node) {
+      const fragmentsInTheSameModule = [];
+      templateLiterals.forEach(templateLiteral => {
+        const graphQLAst = getGraphQLAST(templateLiteral);
```

Comment:
It's not a huge cost to parse the AST, but you could do this once (instead of twice) in your `TaggedTemplateExpression` visitor and store the parsed GraphQL AST along with the eslint node.